src/utils.js

```diff
@@ -0,0 +1,239 @@
+const { execSync } = require("child_process")
+
+module.exports = {
+  countLeaves,
+  dictionary,
+  findKeyInObject,
+  fromAscii,
+  getMaxArgsIndexFromString,
+  isHexString,
+  isHexStringOfLength,
+  isWildcard,
+  mapObjectRecursively,
+  padLeft,
+  parseURL,
+  spliceWildcards,
+  splitSelectionFromProcessArgv,
+
+  dryRunBytecode,
+  dryRunBytecodeVerbose,
+  getRequestMethodString,
+  getRequestResultDataTypeString,
+}
+
+function dictionary(t, dict) {
+  return new Proxy(dict, proxyHandler(t));
+}
+
+function countLeaves(t, obj) {
+  if (!obj) {
+      return 0;
+  }
+  else if (obj instanceof t) {
+      return 1;
+  }
+  else if (Array.isArray(obj)) {
+      return obj.map(function (item) { return countLeaves(t, item); }).reduce(function (a, b) { return a + b; }, 0);
+  }
+  else {
+      return Object.values(obj).map(function (item) { return countLeaves(t, item); }).reduce(function (a, b) { return a + b; }, 0);
+  }
+}
+
+async function dryRunBytecode(bytecode) {
+  return (await execSync(`npx witnet-toolkit try-data-request --hex ${bytecode}`)).toString()
+}
+
+async function dryRunBytecodeVerbose(bytecode) {
+  return (await execSync(`npx witnet-toolkit try-query --hex ${bytecode}`)).toString()
+}
+
+function findKeyInObject(dict, tag) {
+  for (const key in dict) {
+    if (typeof dict[key] === 'object') {
+      if (key === tag) {
+        return dict[key]
+      } else {
+        let found = findKeyInObject(dict[key], tag)
+        if (found) return found
+      }
+    }
+  }
+}
+
+function fromAscii(str) {
+  const arr1 = []
+  for (let n = 0, l = str.length; n < l; n++) {
+    const hex = Number(str.charCodeAt(n)).toString(16)
+    arr1.push(hex)
+  }
+  return "0x" + arr1.join("")
+}
+
+function getMaxArgsIndexFromString(str) {
+  let maxArgsIndex = 0
+  if (str) {
+    let match
+    const regexp = /\\\d\\/g
+    while ((match = regexp.exec(str)) !== null) {
+      let argsIndex = parseInt(match[0][1]) + 1
+      if (argsIndex > maxArgsIndex) maxArgsIndex = argsIndex
+    }
+  }
+  return maxArgsIndex
+}
+
+function getRequestMethodString(method) {
+  if (method == 0) {
+    return "UNKNOWN"
+  } else if (method == 1 || !method) {
+    return "HTTP-GET"
+  } else if (method == 2) {
+    return "RNG"
+  } else if (method == 3) {
+    return "HTTP-POST"
+  } else if (method == 4) {
+    return "HTTP-HEAD"
+  } else {
+    return method.toString()
+  }
```

Comment:
What do you think about using a map here?
```suggestion
  if (!method) {
    return "HTTP-GET
  }

  const methodNameMap = {
    0: "UNKNOWN"
    1: "HTTP-GET"
    2: "RNG"
    3: "HTTP-POST"
    4: "HTTP-HEAD"
   } 
  
  const methodName = methodNameMap[method]
  
  return methodName || method.toString()
```